### ISSUE TITLE
A `NewType`  is reported as "Class created with NewType" when used in `isinstance`

### ISSUE THREAD

**Issue by jorenham** [microsoft/pyright#10601]
```py
A = NewType("A", int)
isinstance(1, A)
```

causes the following error to be reported

```
Second argument to "isinstance" must be a class or tuple of classes
  Class created with NewType cannot be used with instance and class checks  (reportArgumentType)
```

`A` is referred to as the "Class created with NewType", but `A` is not a class.

[pyright-play](https://pyright-play.net/?pyrightVersion=1.1.402&strict=true&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoByApgO4AqiRAUFQIJQC8hpFCRAFAES2cA0mKGAEoqSAM6oxMAIYoAxhwCM-WkKA)

**Comment by jorenham** [comment]
With `class B(A): ...` there's a similar issue:

```
Base class "A" is marked final and cannot be subclassed  (reportGeneralTypeIssues)
```

[pyright-play](https://pyright-play.net/?pyrightVersion=1.1.402&strict=true&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoByApgO4AqiRAUFQIJQC8hpFCRAFAES2cA0mKGAEoqAYwA2AQwDO0qACF2tIQC4oAOk1A)

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.405
